plutus-contract/src/Plutus/Contract/State.hs

```diff
@@ -54,75 +54,85 @@ import           Plutus.Contract.Types            hiding (lastLogs, logs, observ
 
 -- | The state of a 'Contract', containing all responses that have been fed to
 --   it, and checkpoints that it produced.
-data State e = State
-    { record      :: Responses e
-    , checkpoints :: CheckpointStore
+data State w e = State
+    { record          :: Responses e
+    , checkpoints     :: CheckpointStore
+    , observableState :: w
```

Comment:
So this was originally in `ContractResponse`, but I moved it to `State`, because we need it to piece together the `observableState` from the entire execution history of the contract instance, instead of just the last value that was written.